optaplanner-examples/src/main/resources/org/optaplanner/examples/conferencescheduling/solver/conferenceSchedulingScoreRules.drl

```diff
@@ -31,361 +31,405 @@ global HardMediumSoftScoreHolder scoreHolder;
+        $talk2 : Talk(timeslot != null,
+                overlappingContentCount($talk1) > 0,
+                overlapsTime($talk1),
+                id > $id1)
     then
-        scoreHolder.penalize(kcontext, $rightTalk.overlappingContentCount($leftTalk));
+        scoreHolder.penalize(kcontext,
+                $talk2.overlappingContentCount($talk1) * $talk2.overlappingDurationInMinutes($talk1));
 end
 
 rule "Language diversity"
     when
         Talk(timeslot != null, $timeslot : timeslot,
-                $language : language,
-                $leftId : id)
+                $language1 : language,
+                $id1 : id)
         // Timeslot.overlaps() is deliberately not used
         Talk(timeslot == $timeslot,
-                language != $language,
-                id > $leftId)
+                language != $language1,
+                id > $id1)
     then
-        scoreHolder.reward(kcontext);
+        scoreHolder.reward(kcontext,
+                $timeslot.getDurationInMinutes());
 end
 
-rule "Speaker preferred timeslot tags"
-    when
-        Talk($count : missingSpeakerPreferredTimeslotTagCount() > 0)
-    then
-        scoreHolder.penalize(kcontext, $count);
-end
-
-rule "Talk preferred timeslot tags"
+rule "Same day talks"
     when
-        Talk($count : missingPreferredTimeslotTagCount() > 0)
+        $talk1 : Talk(timeslot != null,
+                $id1 : id)
+        $talk2 :  Talk(timeslot != null,
+                overlappingContentCount($talk1) > 0 || overlappingThemeTrackCount($talk1) > 0,
+                !getTimeslot().isOnSameDayAs($talk1.getTimeslot()),
+                id > $id1)
     then
-        scoreHolder.penalize(kcontext, $count);
+        scoreHolder.penalize(kcontext,
+                ($talk2.overlappingThemeTrackCount($talk1) + $talk2.overlappingContentCount($talk1))
+                * ($talk1.getDurationInMinutes() + $talk2.getDurationInMinutes()));
 end
 
-rule "Speaker undesired timeslot tags"
+rule "Popular talks"
     when
-        Talk($count : prevailingSpeakerUndesiredTimeslotTagCount() > 0)
+        $talk1 : Talk(timeslot != null,
+                room != null, $room1 : room,
+                $favoriteCount1 : favoriteCount)
+        $talk2 : Talk(timeslot != null,
+                room != null,
+                favoriteCount < $favoriteCount1,
+                getRoom().getCapacity() > $room1.getCapacity())
     then
-        scoreHolder.penalize(kcontext, $count);
+        scoreHolder.penalize(kcontext,
+                $talk1.getDurationInMinutes() + $talk2.getDurationInMinutes());
 end
 
-rule "Talk undesired timeslot tags"
+rule "Crowd control"
     when
-        Talk($count : prevailingUndesiredTimeslotTagCount() > 0)
+        $talk : Talk(timeslot != null, $timeslot : timeslot,
+                crowdControlRisk > 0)
+        Number(this != 1) from accumulate(
+            // Timeslot.overlaps() is deliberately not used
+            $otherTalk : Talk(timeslot == $timeslot,
```

Comment:
If the intention behind not using Timeslot.overlaps() is that crowd control affects attendees' decision only if two timeslots start at the same time, I thnk it makes more sense to use timeslot.getStartDateTime().equals($timeslot.getStartDateTime())

Reply:
Good point. It's weird, but the end time indeed doesn't matter. There are far less people going jump in the middle of an ongoing session.

Crowd control is defined by Stephan for Devoxx, and Devoxx doesn't have overlapping timeslots (mostly anyway), so this is uncharted territory - hard to define the correct behavior here. Let's make a jira and postpone.

Reply:
I've added this todo in the drl:
            // TODO Analyse if this is better: getTimeslot().getStartDateTime().equals($timeslot.getStartDateTime())
            // because from a crowd control perspective, only the start of the session matters